sql/sql_explain.cc

```diff
@@ -351,9 +351,9 @@ int print_explain_row(select_result_sink *result,
                       mem_root);
   
   /* 'possible_keys' */
+  StringBuffer<64> possible_keys_buf;
   if (possible_keys && !possible_keys->is_empty())
   {
-    StringBuffer<64> possible_keys_buf;
```

Comment:
possible_keys_buf can be allocated either on stack or on heap. While corresponding Item_string_sys is allocated on thd->mem_root. This means that they have different lifetime. Not sure if there's a real problem, but this all sounds fragile.